cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -109,9 +109,10 @@ open class Node @Inject constructor(private val project: Project) {
      *
      * @return This node's P2P address.
      */
-    val p2pAddress: String
+    @get:Optional
+    val p2pAddress: String?
         @Input
-        get() = config.getString("p2pAddress")
```

Comment:
Pick _one_ convention for applying these annotations, and then stick with it?
```kotlin
val p2pAddress: String?
    @Optional
    @Input
    get() = ...
```
You could even write _this_, if you were feeling _particularly_ perverse...
```kotlin
@get:[Optional Input]
val p2pAddress: String?
    get() = ...
```